File: src/main/java/com/webappdevelopment/service/DealRepService.java
package com.webappdevelopment.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.webappdevelopment.domain.DealRep;
import com.webappdevelopment.dto.DealRepDTO;
import com.webappdevelopment.dto.DealRepSearchDTO;
import com.webappdevelopment.dto.DealRepPageDTO;
import com.webappdevelopment.dto.DealRepConvertCriteriaDTO;
import com.webappdevelopment.service.GenericService;
import com.webappdevelopment.dto.common.RequestDTO;
import com.webappdevelopment.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface DealRepService extends GenericService<DealRep, Integer> {

	List<DealRep> findAll();

	ResultDTO addDealRep(DealRepDTO dealRepDTO, RequestDTO requestDTO);

	ResultDTO updateDealRep(DealRepDTO dealRepDTO, RequestDTO requestDTO);

    Page<DealRep> getAllDealReps(Pageable pageable);

    Page<DealRep> getAllDealReps(Specification<DealRep> spec, Pageable pageable);

	ResponseEntity<DealRepPageDTO> getDealReps(DealRepSearchDTO dealRepSearchDTO);
	
	List<DealRepDTO> convertDealRepsToDealRepDTOs(List<DealRep> dealReps, DealRepConvertCriteriaDTO convertCriteria);

	DealRepDTO getDealRepDTOById(Integer dealRepId);







}

File: webappdevelopment_ui/src/services/DealRepService.js
import http from "../http-common"; 

class DealRepService {
  getAllDealReps(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/dealRep/dealReps`, searchDTO);
  }

  get(dealRepId) {
    return this.getRequest(`/dealRep/${dealRepId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/dealRep?field=${matchData}`, null);
  }

  addDealRep(data) {
    return http.post("/dealRep/addDealRep", data);
  }

  update(data) {
  	return http.post("/dealRep/updateDealRep", data);
  }
  
  uploadImage(data,dealRepId) {
  	return http.postForm("/dealRep/uploadImage/"+dealRepId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new DealRepService();
